demonstrations/tutorial_optimal_control.py

```diff
@@ -0,0 +1,694 @@
+    ahs_aquila Pulse programming on Rydberg atom hardware
+
+*Author: David Wierichs. Posted: xx June, 2023.*
+
+
+Today, quantum computations largely are phrased as quantum circuits--or gate 
+sequences more generally--that are composed of digital quantum gates.
+However, most quantum hardware does not come with such digital
+gates as elementary native operations.
+Instead, it allows us to play sequences of analogue electromagnetic pulses,
+for example laser pulses that interact with trapped ions or Rydberg atoms, or microwave
+pulses acting on superconducting qubits.
+These pulses need to be calibrated to yield the desired digital gates, and in
+this tutorial we will be concerned with exactly that step.
+For this, we parametrize a pulse sequence, which yields a whole *space* 
+of sequences, and then optimize the pulse parameters to find a configuration
+in the space that acts as closely to the gate we are interested in
+as possible. This training of control parameters to achieve a specific time
+evolution falls under the field of *quantum optimal control*.
+
+More concretely, we will optimize simple pulse programs on two and three qubits to
+obtain a CNOT and a Toffoli gate. 
+
+For an introduction see
+:doc:`the demo on differentiable pulse programming </demos/tutorial_pulse_programming101>`
+in PennyLane.
+Instead of optimizing pulses to yield digital gates that are used in quantum circuits,
+we may use them directly to solve minimization problems, as is also showcased in the demo.
+If you are interested in specific hardware pulses, take a look at
+:doc:`an introduction to neutral-atom quantum computing </demos/tutorial_neutral_atoms>`
+or :doc:`the tutorial on the QuEra Aquila device </demos/ahs_aquila>`, which treat pulse
+programming with Rydberg atoms.
+
+Quantum optimal control
+-----------------------
+
+The overarching goal of quantum optimal control is to find the best way to steer
+a microscopical physical system such that its dynamics matches a desired behaviour.
+The meaning of "best" and "desired behaviour" will depend on the specific
+task, and it is important to specify underlying assumptions and constraints on
+the system controls in order to make the problem statement well-defined.
+Once we specified all these details, optimal control theory is concerned with
+questions like
+"How close can the system get to modelling the desired behaviour?",
+"How can we find the best (sequence of) control parameters to obtain the desired behaviour?",
+or
+"What is the shortest time in which the system can reach a specific state, given some
+initial state?".
+
+In this tutorial, we consider the control of few-qubit systems through pulse sequences,
+with the goal to produce a given target, namely a digital gate, to the highest-possible
+precision.
+To do this, we will choose an ansatz for the pulse sequence that contains
+free parameters and define a cost function that measures the deviation of the qubit
+system from the target gate.
```

Comment:
not sure about "to measure" and "qubit system"
```suggestion
free parameters and define a cost function that quantifies the deviation of the qubit
operation from the target gate.
```